frontend/src/features/admin-form/create/logic/hooks/useAdminFormLogic.ts

```diff
@@ -21,11 +23,29 @@ export const useAdminFormLogic = () => {
     return pickBy(mapIdToField, (f) => ALLOWED_LOGIC_FIELDS.has(f.fieldType))
   }, [mapIdToField])
 
+  const hasError = useMemo(
+    () =>
+      !mapIdToField
+        ? false
+        : form?.form_logics.some(
+            (logic) =>
+              // Logic is errored if some condition does not exist, or all the
+              // show fields do not exist.
+              logic.conditions.some(
+                (condition) => !(condition.field in mapIdToField),
+              ) ||
+              (logic.logicType === LogicType.ShowFields &&
+                logic.show.every((field) => !(field in mapIdToField))),
+          ),
```

Comment:
double negatives are a bit confusing
```suggestion
    () => {
      if (!mapIdToField || !form) return false
      return form.form_logics.some(
         // etc
      )
    }
    }
```

Reply:
will change!